### PR TITLE
Add QueryExecutor#inNewConnection method

### PR DESCRIPTION
**Pull request by findepi**
Method name was picked based on JDBI naming convention.

### CODE REVIEW

File: tempto-core/src/main/java/io/prestosql/tempto/query/JdbcQueryExecutor.java [hunk 1]
```diff
@@ -92,6 +98,16 @@ public Connection getConnection()
         return connection;
     }
 
+    @Override
+    public <T> T inNewConnection(ConnectionCallback callback)
```
Comment (kokosing) [inline tempto-core/src/main/java/io/prestosql/tempto/query/JdbcQueryExecutor.java]: `withConnection`? or `withNewConnection`?

Other than I thought we already have something like that.

**Review by kokosing — APPROVED** [review]

**Review by kokosing — COMMENTED** [review]
Could give a simple test (in tempto-examples).

File: tempto-core/src/main/java/io/prestosql/tempto/query/JdbcQueryExecutor.java (hunk 1 above)
Comment (findepi) [reply tempto-core/src/main/java/io/prestosql/tempto/query/JdbcQueryExecutor.java]: > `withConnection`? or `withNewConnection`?

JDBI uses `useConnection` (for `void` returning methods) and `inConnection` for value-bearing methods

> Other than I thought we already have something like that.

 do we?
we have `getConnection()`

Comment (kokosing) [reply tempto-core/src/main/java/io/prestosql/tempto/query/JdbcQueryExecutor.java]: > do we?

Unfortunately no.

**Pull request closed without merging**